### DISCUSSION TITLE
Minimum physical connections required to program the microcontroller?

### DISCUSSION THREAD

**Discussion by mchopping** [STELLA-Landsat/STELLA#31] [Build Notes: How do I make a STELLA?]
I wonder whether it's possible to program the microcontroller -- at least "Step 1: Programming STELLA-Q microcontroller", if not all three steps -- while having as few components attached as possible. The idea is that my students can handle the physical construction but the programming steps are likely to be difficult for them to follow and this also requires specialist software we'd have to have installed in our GIS lab, so it may make sense for the instructor (me) to do as much of the programming as possible in advance, without having to build and deconstruct. Of course, setting the time / RTC testing can't work unless the RTC is attached via the hub. Or perhaps it is better to get the software  installed and have the students wrestle with the programming after they have put their STELLA-Qs together? Does anyone have any advice, or experience with this?

**Comment by paulmirel** [comment]
That is a very good way to proceed. Do the arcane work of loading CircuitPython, and copy the code and lib folder onto the processors, so the students don't have to learn to do these steps. If you populate the batteries for the clock modules ahead of time, you can also set all the clock modules. That will put the kids in a position to be able to use the instrument right after making the connections. I also charge up the main batteries before a lesson, so everything is ready.

**Reply by mchopping** [reply]
Thanks Paul. I assume you are confirming it is possible to do this with just the RTC and hub --and power-- attached, is that right?

**Comment by paulmirel** [comment]
And, yes, it is entirely possible to program the microcontroller without connecting any of the other devices.

**Reply by mchopping** [reply]
All questions answered, thanks!
